### ISSUE TITLE
Sauron disappears after teleport level

### ISSUE THREAD

**Issue by HughWarrington** [angband/angband#4708]
I am playing Angband 4.2.1 on Windows.

I got to dungeon level 99 (not for the first time on this run) and began fighting Sauron.

He transformed into Serpent-Sauron and a couple of rounds later I got

```
Serpent-Sauron commands you to go far away.
You rise up through the ceiling.
Serpent-Sauron speeds up.
This place seems terribly dangerous.
```

Now when I go back down to level 99, there's no sign of Sauron, and also no down staircase. I've been up and down levels multiple times and conducted a thorough search of three different level 99s.

Is this the intended behaviour? It feels like a bug to me, based on what I can tell from the forums (and also my memories of playing many versions of Angband over the years).

**Comment by backwardsEric** [comment]
That was one of the issues that's believed to be fixed in 4.2.2; this change, https://github.com/angband/angband/commit/e9d8b79f8c6d40c6ffd28e6406327be119206734#diff-4d8476305de0d2d06259216edd584fe20cf6389d8958e91f66a6a5a84290d0b4 , is the most relevant:  leaving a non-persistent level with a shape-changed unique on the level marks that unique as capable of appearing again on another level.

If you're still interested in completing the game from that save file, posting the compressed save file (and if you are using randarts the related randart file from the archive directory), should allow us to do some surgery on that save file and allow Sauron to reappear.

**Comment by NickMcConnell** [comment]
I'm assuming this can be closed now.
